java/code/src/com/redhat/rhn/frontend/xmlrpc/recurringaction/RecurringActionHandler.java

```
@@ -0,0 +1,243 @@
+     *      #prop_desc("boolean", "test", "Whether the action should be executed in test mode (optional)")
+     *      #prop_desc("boolean", "active", "Whether the action should be active (optional)")
+     *  #struct_end()
+     * @xmlrpc.returntype int action_Id - The action id of the recurring action
+     */
+    public Long update(User loggedInUser, Map<String, Object> actionProps) {
+        RecurringAction action = updateAction(actionProps, loggedInUser);
+        return save(loggedInUser, action);
+    }
+
+    /* Helper method */
+    private  RecurringAction updateAction(Map<String, Object> actionProps, User user) {
+        if (!actionProps.containsKey("id")) {
+            throw new InvalidArgsException("No action id provided");
+        }
+        Integer id = Integer.parseInt((String) actionProps.get("id"));
+        RecurringAction action = lookupById(user, id);
+
+        if (actionProps.containsKey("name")) {
+            action.setName((String) actionProps.get("name"));
+        }
+        if (actionProps.containsKey("cron_expr")) {
+            action.setCronExpr((String) actionProps.get("cron_expr"));
+        }
+        if (actionProps.containsKey("test")) {
+            action.setTestMode(Boolean.parseBoolean(actionProps.get("test").toString()));
+        }
+        if (actionProps.containsKey("active")) {
+            action.setActive(Boolean.parseBoolean(actionProps.get("active").toString()));
+        }
+        return action;
+    }
+
+    /* Helper method */
+    private Long save(User loggedInUser, RecurringAction action) {
+        try {
+            RecurringActionManager.saveAndSchedule(action, loggedInUser);
+        }
+        catch (ValidatorException e) {
+            throw new ValidationException(e.getMessage());
+        }
+        catch (com.redhat.rhn.taskomatic.TaskomaticApiException e) {
+            HibernateFactory.rollbackTransaction();
```

I've just checked and it seems that this line is not needed. The XMLRPC backend seems **not** to swallow certain exceptions (the UI does it, apparently), so if we just rethrow the exception on the line below, the transaction will be rolled back automatically :)